File: back/src/menus/entity/menu.entity.ts
import {
  BaseEntity,
  Column,
  Entity,
  JoinColumn,
  OneToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { MealType } from '../enum/meal-type.enum';
import { MenuStats } from './menu-stats.entity';

@Entity()
export class Menu extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column('varchar', { length: 255 })
  name: string;

  @Column('text')
  description: string;

  @Column('int')
  calories: number;

  @Column({
    type: 'enum',
    enum: MealType,
    default: MealType.LUNCH,
  })
  mealType: MealType;

  @Column('text', { nullable: true })
  imageUrl: string;

  @Column({ type: 'boolean', default: false })
  isLunchBox: boolean;

  @Column({ type: 'date' })
  date: Date;

  @OneToOne(() => MenuStats)
  @JoinColumn()
  stats: MenuStats;
}

File: back-spring/src/main/java/com/nhn_eat/back/entity/MenuEntity.java
package com.nhn_eat.back.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Enumerated;
import jakarta.persistence.EnumType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "menu")
@Setter
@Getter
@ToString
@EqualsAndHashCode(exclude = "id")
public class MenuEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String description;

    @Column(nullable = true)
    private int calories;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MealType mealType;

    @Column(nullable = true)
    private String imageUrl;

    @Column(nullable = false)
    private boolean isLunchBox;

    @Column(nullable = false)
    private String date;
}
